### PR TITLE
Enable creation of Command NuGet packages via dotnet-pack

### PR DESCRIPTION
**Pull request by piotrpMSFT**
Fixes issue #365

### CODE REVIEW

File: src/Microsoft.DotNet.Tools.Pack/Program.cs [hunk 1]
```diff
@@ -109,15 +114,19 @@ private static bool BuildPackage(string path, string configuration, string outpu
 
                 if (result.ExitCode != 0)
                 {
-                    return false;
+                    throw new InvalidOperationException($"dotnet-compile returned a non-0 exit code: '{result.ExitCode}'.");
                 }
             }
+        }
+
+        private static void BuildPackage(IEnumerable<ProjectContext> contexts, string configuration, string outputValue)
```
Comment (krwq) [inline src/Microsoft.DotNet.Tools.Pack/Program.cs]: we got a conflict here, If my change happens to get in first just take yours

File: src/Microsoft.DotNet.Tools.Pack/project.json [hunk 2]
```diff
@@ -27,5 +27,11 @@
   },
   "frameworks": {
     "dnxcore50": { }
+  },
+  "scripts": {
+    "postcompile": [
+      "../../scripts/build/place-binary \"%compile:OutputDir%/%project:Name%.dll\"",
```
Comment (krwq) [inline src/Microsoft.DotNet.Tools.Pack/project.json]: aren't we moving away from post steps?

**Comment by krwq** [comment]
Nice PR! Looks good to me :+1: :ship:

File: src/Microsoft.DotNet.Tools.Pack/project.json (hunk 2 above)
Comment (krwq) [inline src/Microsoft.DotNet.Tools.Pack/project.json]: we should probably clean it up separately

**Comment by piotrpMSFT** [comment]
@dotnet-bot test this please

**Comment by krwq** [comment]
everything seems good :shipit:

**Comment by davidfowl** [comment]
Anything you do here is short term FYI. We're currently designing how these packages are going to look and work with nuget.exe xplat.

File: src/Microsoft.DotNet.Tools.Pack/ProjectExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.ProjectModel;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Tools.Pack
+{
+    public static class ProjectExtensions
+    {
+        public static bool IsCommand(this Project project, NuGetFramework targetFramework, string configurationName)
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Pack/ProjectExtensions.cs]: This is hella hacky. Why not just use the EmitEntryPoint field?

**Comment by krwq** [comment]
@davidfowl, what timeframe are we talking? Will this happen before January?

**Comment by piotrpMSFT** [comment]
We should have some updates by week's end.

**Comment by piotrpMSFT** [comment]
We took a different approach here.

**Pull request closed without merging**